ast/misc.go

```diff
@@ -1045,7 +1045,20 @@ type TableOptimizerHint struct {
 
 // Restore implements Node interface.
 func (n *TableOptimizerHint) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord(n.HintName.String())
+	ctx.WritePlain("(")
+	if n.HintName.O == "MAX_EXECUTION_TIME" {
```

Comment:
```suggestion
	if n.HintName.L == "max_execution_time" {
```

I suppose `/*+ max_execution_time(3000) */` is a valid hint.